commands/certs/add.js

```diff
@@ -90,32 +120,46 @@ function * addDomains (context, heroku, meta, cert) {
     existingDomains.forEach((domain) => cli.log(domain))
   }
 
-  let addedDomains
-  if (newDomains.length > 0) {
-    let choices
-    if (context.flags.domains) {
-      choices = getFlagChoices(context, certDomains, existingDomains)
-    } else {
-      choices = (yield getPromptChoices(context, certDomains, existingDomains, newDomains)).domains
-    }
+  let choices = yield getChoices(certDomains, newDomains, existingDomains, context)
+  let domains
 
+  if (choices.length === 0) {
+    domains = new Domains([])
+  } else {
     // Add a newline between the existing and adding messages
-    if (choices.length > 0) {
-      cli.console.error()
-    }
+    cli.console.error()
 
-    addedDomains = new Array(choices.length)
-    for (let i = 0; i < choices.length; i++) {
-      let certDomain = choices[i]
-
-      addedDomains[i] = yield cli.action(`Adding domain ${certDomain} to ${context.app}`, {}, heroku.request({
+    let promise = Promise.all(choices.map(function (certDomain) {
+      return heroku.request({
         path: `/apps/${context.app}/domains`,
         method: 'POST',
         body: {'hostname': certDomain}
-      }))
-    }
-  } else {
-    addedDomains = []
+      }).catch(function (err) {
+        return {_hostname: certDomain, _failed: true, _err: err}
+      })
+    })).then(function (data) {
+      let domains = new Domains(data)
+      if (domains.hasFailed) {
+        throw domains
+      }
+      return domains
+    })
+
+    let label = choices.length > 1 ? 'domains' : 'domain'
+    domains = yield cli.action(`Adding ${label} ${choices.join(', ')} to ${context.app}`, {}, promise).catch(function (err) {
+      if (err instanceof Domains) {
+        return err
+      }
+      throw err
+    })
+  }
+
+  if (domains.hasFailed) {
```

Comment:
@dickeyxxx It will only exit non-zero at this point if we get an error unrelated to the domains add.  Should this exit non-zero if the certificate was added, but some of the domains were unable to be added.  I am leaning toward non-zero exit if a domain add failed.